Add status() docs to Client JS reference (#30)
web3-storage/docs: docs/reference/client-library.md

| Number | Type | Description |
| ------ | ---- | ----------- |
| 1 | `file[]` | An iterable collection of [Files](https://developer.mozilla.org/en-US/docs/Web/API/File) to be packed into a CAR and uploaded. |

> | 2 | `{options}` | Optional. A JSON object whose properties define certain Web3.Storage options and metadata about the files being uploaded. See below for more details. |

Reviewer: Not JSON, just a plain object.

```suggestion
| 2 | `{options}` | Optional. An object whose properties define certain Web3.Storage options and metadata about the files being uploaded. See below for more details. |
```

| 2 | `{options}` | Optional. A JSON object whose properties define certain Web3.Storage options and metadata about the files being uploaded. See below for more details. |

An `{options}` object is a JSON object that defines option parameters:

```js
```json
{
  name: 'cat pics',
  name: "cat pics",
  maxRetries: 3,
  onRootCidReady: handlerFunction,

>   onStoredChunk: handlerFunction

Reviewer: You can find good descriptions of these options here: https://github.com/web3-storage/web3.storage/blob/20d02fc22725de825cc2a4f773b40e0d1f8a6a68/packages/client/src/lib/interface.ts#L62

Author: Just a bit further down the parameters are covered in much better detail so I removed this bit

| ------ | ---- | ----------- |
| 1 | `string` | A string containing the CID of the CAR to be retrieved. |

## Check status

> Retrieve metadata about your file by using the `status()` method, supplying the CID of the file you are interested in. This metadata includes the creation date and size, as well as details about how the network is storing your data. Using this information you can enumerate peers on the InterPlanetary File System (IPFS) network that are pinning the data, and Filecoin miners that have accepted deals to store the data.

Reviewer: ```suggestion
Retrieve metadata about your file by using the `status()` method, supplying the CID of the file you are interested in. This metadata includes the creation date and size, as well as details about how the network is storing your data. Using this information you can enumerate peers on the InterPlanetary File System (IPFS) network that are pinning the data, and Filecoin storage providers that have accepted deals to store the data.
```

### Return value

> The `status()` method returns a `{Status}` object that contains the metadata for your object's storage deal on the Web3.Storage network. This object will have the following properties:

Reviewer 2: I think we should mention that it will return `undefined` if the CID isn't found.

Reviewer 3: Aye! I'm not sure of the ergonomics of that decision tho. We could just have it throw on anything other than a Cid that we were able to get a status for. What do you think?

Author: @olizilla I feel like returning an object with really obvious "not found" values might be more merciful given that `undefined` will almost surely throw people's code into an exception state, but hey.

Reviewer 2: I like the idea of a "not found" response object a bit better than throwing, but either way could work. For now it seems worth documenting the current behavior & we can update if the API changes.

Reviewer 3: For context, I took my cues from the behaviour of `Map.get()` here. It returns `undefined` if you ask for a thing that is not in it. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get

Author: Agreed on doc'ing current behavior. Added details for `undefined` for `status()` and `get()`. 

I ain't mad at the method returning `undefined` -- stark as it is. I just have a hunch that a "not found" response object with a helpful message embedded is somewhat more akin to, for instance, an HTTP status code of 500 with details in the payload. Food for thought, anyway. :)
